android-lib/src/androidTest/java/com/kinvey/androidTest/store/data/DataStoreMultiInsertTestKotlin.kt

```diff
@@ -0,0 +1,1954 @@
+        val looperThread = LooperThread(Runnable { store.delete(query, callback) })
+        looperThread.start()
+        latch.await(120, TimeUnit.SECONDS)
+        looperThread.mHandler.sendMessage(Message())
+        return callback
+    }
+
+    @Throws(InterruptedException::class)
+    private fun <T : GenericJson> find(store: DataStore<T>, seconds: Int): DefaultKinveyReadCallback<T> {
+        val latch = CountDownLatch(1)
+        val callback = DefaultKinveyReadCallback<T>(latch)
+        val looperThread = LooperThread(Runnable { store.find(callback) })
+        looperThread.start()
+        latch.await()
+        looperThread.mHandler.sendMessage(Message())
+        return callback
+    }
+
+    @Throws(InterruptedException::class)
+    private fun sync(store: DataStore<Person>, seconds: Int): DefaultKinveySyncCallback {
+        val latch = CountDownLatch(1)
+        val callback = DefaultKinveySyncCallback(latch)
+        val looperThread = LooperThread(Runnable { store.sync(callback) })
+        looperThread.start()
+        latch.await(seconds.toLong(), TimeUnit.SECONDS)
+        looperThread.mHandler.sendMessage(Message())
+        return callback
+    }
+
+    @Throws(InterruptedException::class)
+    private fun <T : GenericJson> clearBackend(store: DataStore<T>) {
+        var query = client?.query()
+        query = query?.notEqual("age", "100500")
+        query?.let { delete(store, query) }
+    }
+
+    private fun pendingSyncEntities(collectionName: String): List<SyncItem>? {
+        return client?.syncManager?.popSingleItemQueue(collectionName)
+    }
+
+    private fun print(msg: String) {
+        Logger.INFO(msg)
+        println(msg)
+    }
+
+    @Throws(InterruptedException::class)
+    private fun <T : GenericJson> saveList(store: DataStore<T>, persons: List<T>): DefaultKinveyClientListCallback<T> {
+        val latch = CountDownLatch(1)
+        val callback = DefaultKinveyClientListCallback<T>(latch)
+        val looperThread = LooperThread(Runnable {
+            try {
+                store.save(persons, callback)
+            } catch (e: Exception) {
+                callback.onFailure(e)
+            }
+        })
+        looperThread.start()
+        latch.await()
+        looperThread.mHandler.sendMessage(Message())
+        return callback
+    }
+
+    @Throws(InterruptedException::class)
+    private fun <T : GenericJson> save(store: DataStore<T>, item: T): DefaultKinveyClientCallback<T> {
+        val latch = CountDownLatch(1)
+        val callback = DefaultKinveyClientCallback<T>(latch)
+        val looperThread = LooperThread(Runnable { store.save(item, callback) })
+        looperThread.start()
+        latch.await()
+        looperThread.mHandler.sendMessage(Message())
+        return callback
+    }
+
+    @Throws(InterruptedException::class)
+    private fun <T : GenericJson> push(store: DataStore<T>, seconds: Int): DefaultKinveyPushCallback {
```

Comment:
let's move those methods to a base class and try to keep only methods annotated with `@Test` here

Reply:
Moved common code base class `BaseDataStoreTest`.